### PR TITLE
Library app 3: Add List of Books & Form as separate component

### PR DESCRIPTION
**Pull request by bhoomika08**

### CODE REVIEW

File: LibraryApp/src/components/books-list.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+          <Text style={bookPrice}>{price}</Text>
+        </View>
+      </Pressable>
+    );
+  };
+
+  getItemKey = (item) => `book-${item.id}`;
+
+  search(value) {
+    this.setState({
```
Comment (gash48) [inline LibraryApp/src/components/books-list.js]: this is inhibiting user behaviour.Search should be trimmed in serach logic not here

File: LibraryApp/src/components/books-list.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    }, loaderTimeout);
+  }
+
+  onRefresh() {
+    this.setState({isRefreshing: true}, () => {
+      this.getApiData();
+    });
+  }
+
+  getApiData() {
```
Comment (gash48) [inline LibraryApp/src/components/books-list.js]: Used once

File: LibraryApp/src/components/books-list.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+          <Text style={[gochiFont, fs18]}>{publisher}</Text>
+        </View>
+        <View>
+          <Text style={bookPrice}>{price}</Text>
+        </View>
+      </Pressable>
+    );
+  };
+
+  getItemKey = (item) => `book-${item.id}`;
```
Comment (gash48) [inline LibraryApp/src/components/books-list.js]: Not needed inside component

File: LibraryApp/src/components/books-list.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    this.search = this.search.bind(this);
+    this.addNewBook = this.addNewBook.bind(this);
+  }
+
+  componentDidMount() {
+    const {books} = this.props;
+    if (books.length == 0) {
+      this.setState({
+        isDataFetching: true,
+      });
```
Comment (gash48) [inline LibraryApp/src/components/books-list.js]: Call timeout in callback

File: LibraryApp/src/components/books-list.js [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    });
+  }
+
+  getApiData() {
+    setTimeout(() => {
+      this.setState({isRefreshing: false});
+    }, loaderTimeout);
+  }
+
+  selectBook(book) {
```
Comment (gash48) [inline LibraryApp/src/components/books-list.js:56]: Select Boook & Add Book are same

File: LibraryApp/src/components/books-list.js [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    const filteredList = FilterListBySearch(searchText, booksArr);
+    const containsBooks = filteredList.length > 0;
+    return (
+      <>
+        <View style={titleContainer}>
+          <Text style={title}>List of the Books</Text>
+        </View>
+        <View style={p15}>
+          <View style={searchContainer}>
+            <View style={[verticalCenter]}>
```
Comment (gash48) [inline LibraryApp/src/components/books-list.js]:
```suggestion
            <View style={verticalCenter}>
```

File: LibraryApp/src/components/books-list.js [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+              <Text style={searchIcon}>{search}</Text>
+            </View>
+            <TextInput
```
Comment (gash48) [inline LibraryApp/src/components/books-list.js:122]: add a ReturnKeyType for input fields

File: LibraryApp/src/components/form.js [hunk 8]
```diff
@@ -97,36 +127,24 @@ class Form extends React.PureComponent {
...
+  onSuccess(bookData) {
+    const {setActiveTab} = this.props;
     this.resetAllFields();
-    alert('Success', stringifyResponse(response));
-  }
-
-  onError(error) {
-    this.setState({isLoading: false});
-    alert('Error', error.message);
+    alert('Book Details added / updated', stringifyResponse(bookData));
```
Comment (gash48) [inline LibraryApp/src/components/form.js]: Dont stringify response in message

File: LibraryApp/src/components/form.js [hunk 9]
```diff
@@ -151,23 +169,26 @@ class Form extends React.PureComponent {
...
-      <KeyboardAvoidingView
-        style={innerContainer}
-        behavior={isIOSPlatform ? 'padding' : null}
-        keyboardVerticalOffset={0}>
-        <View style={headingContainer}>
-          <Text style={formLabel}>Library Form</Text>
+      <>
+        <View style={backNavigationContainer}>
+          <Pressable style={rowFlex} onPress={this.displayBooksList}>
+            <View style={[verticalCenter]}>
```
Comment (gash48) [inline LibraryApp/src/components/form.js]:
```suggestion
            <View style={verticalCenter}>
```

File: LibraryApp/src/components/shared/form-controls.js [hunk 10]
```diff
@@ -100,33 +108,33 @@ export const Dropdown = ({
     iosDropdownContainer,
     picker,
     dropdown,
+    dropdownItem,
   } = styles;
-  const selectedValue = options?.[selectedId]?.value;
+
+  const selectedVal = selectedValue;
```
Comment (gash48) [inline LibraryApp/src/components/shared/form-controls.js]: is this necessary ?

File: LibraryApp/src/components/library-app.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+const isIOSPlatform = Platform.OS == 'ios';
+
+class LibraryApp extends React.PureComponent {
+  constructor() {
+    super();
+    this.activateList = this.activateList.bind(this);
+    this.activateForm = this.activateForm.bind(this);
+  }
+
+  activateList() {
```
Comment (gash48) [inline LibraryApp/src/components/library-app.js]: are these used ?

File: LibraryApp/src/components/library-app.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    const {setActiveTab} = this.props;
+    setActiveTab(booksList);
+  }
+
+  activateForm() {
```
Comment (gash48) [inline LibraryApp/src/components/library-app.js]: are these used ?

File: LibraryApp/src/constants/icons.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
+export const Icons = {
```
Comment (gash48) [inline LibraryApp/src/constants/icons.js]: change import style

File: LibraryApp/src/components/books-list.js (hunk 5 above)
Comment (bhoomika08) [reply LibraryApp/src/components/books-list.js:56]: Both are different. Avoiding arrow function.

File: LibraryApp/src/components/books-list.js (hunk 1 above)
Comment (bhoomika08) [reply LibraryApp/src/components/books-list.js]: Done

File: LibraryApp/src/components/books-list.js (hunk 2 above)
Comment (bhoomika08) [reply LibraryApp/src/components/books-list.js]: removed this method.

File: LibraryApp/src/components/books-list.js (hunk 3 above)
Comment (bhoomika08) [reply LibraryApp/src/components/books-list.js]: Done

File: LibraryApp/src/components/books-list.js (hunk 4 above)
Comment (bhoomika08) [reply LibraryApp/src/components/books-list.js]: not used now. so removed it

File: LibraryApp/src/components/books-list.js (hunk 6 above)
Comment (bhoomika08) [reply LibraryApp/src/components/books-list.js]: done

File: LibraryApp/src/components/form.js (hunk 8 above)
Comment (bhoomika08) [reply LibraryApp/src/components/form.js]: Removed bookData. Not displaying submitted book details on alert.

File: LibraryApp/src/components/form.js (hunk 9 above)
Comment (bhoomika08) [reply LibraryApp/src/components/form.js]: Done

File: LibraryApp/src/components/shared/form-controls.js (hunk 10 above)
Comment (bhoomika08) [reply LibraryApp/src/components/shared/form-controls.js]: Nope. Removed.

File: LibraryApp/src/components/library-app.js (hunk 11 above)
Comment (bhoomika08) [reply LibraryApp/src/components/library-app.js]: Deleted.

File: LibraryApp/src/components/library-app.js (hunk 12 above)
Comment (bhoomika08) [reply LibraryApp/src/components/library-app.js]: Deleted.

File: LibraryApp/src/components/books-list.js (hunk 7 above)
Comment (bhoomika08) [reply LibraryApp/src/components/books-list.js:122]: Added returnKeyType="search" but searching is working on entering search text only.

**Pull request merged**